File: Contents/InitialConsonantSearch/InitialConsonantSearch/Main/TEST/Cell/ToolSettingCell.m
//
//  TableViewCell.m
//  ChartTypeTest
//
//  Created by Kwan Hyun Son on 2023/08/31.
//

#import "ToolSettingCell.h"

@interface ToolSettingCell ()
@property (weak, nonatomic) IBOutlet UIButton *titleLabel;
@end

@implementation ToolSettingCell

#pragma mark - Override

- (void)awakeFromNib {
    [super awakeFromNib];
    self.titleLabel.backgroundColor = [UIColor clearColor];
    self.titleLabel.userInteractionEnabled = NO;
    self.showsReorderControl = YES;
    self.backgroundColor = [UIColor whiteColor]; // 드래그 시 그림자를 나오게 해야한다
}

#pragma mark - 세터 & 게터

- (void)setTitle:(NSString *)title {
    _title = title;
    [self.titleLabel setTitle:title forState:UIControlStateNormal];
}

@end

File: Contents/InitialConsonantSearch/InitialConsonantSearch_Swift/Main/TEST/Cell/ToolSettingCell.swift
//
//  ChartTypeCell.swift
//  ChartType_Swift
//
//  Created by Kwan Hyun Son on 2023/09/08.
//

import UIKit

final class ToolSettingCell: UITableViewCell {
    
    // MARK: - Property
    
    var title: String? {
        didSet {
            titleLabel.setTitle(title, for: .normal)
        }
    }
    
    @IBOutlet private weak var titleLabel: UIButton!
    
    // MARK: - Override
    
    override func awakeFromNib() {
        super.awakeFromNib()
        titleLabel.backgroundColor = .clear
        titleLabel.isUserInteractionEnabled = false
        showsReorderControl = true
        backgroundColor = .white // 드래그 시 그림자를 나오게 해야한다
    }
}
